### PR TITLE
[App Service] Fix #26736: `az logicapp create`: Add `--runtime-version` and `--functions-version` optional parameters

### LINKED ISSUE #26736: "az logicapp create" creates Logic App with currently unsupported application setting parameter
### Related command

az logicapp create --subscription {subscriptionId} -g {resourceGroup} -n {name of Logic App Stanard} -p {name or resourceId of App Service Plan} -s {name of storage account} --disable-app-insights true

### Resource Provider

Microsoft.Web/sites

### Description of Feature or Work Requested

"az logicapp create" creates now Logic App with following 3 unsupported application setting parameter.

1. FUNCTIONS_EXTENSION_VERSION which has value "~3". Azure Functions version v3 is not currently supported. Recommended value is "~4".

2. There is not WEBSITE_NODE_DEFAULT_VERSION parameter. This parameter needs to exist and recommended value is currently "~16" in Logic Apps Standard. (Function App recommends to be used value "~18".)

3. There is MACHINEKEY_DecryptionKey. This value is for .NET and Logic Apps Standard is used with node.js. This parameter shoudn't exist in Logic Apps Standard.

### Minimum API Version Required

2015-08-01

### Swagger PR link / SDK link

N/A

### Request Example

_No response_

### Target Date

N/A

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by archidda**
… create

**Related command**

**Description**

**Testing Guide**

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
az logicapp

**Comment by yonzhan** [comment]
This PR is still in Draft version.

**Review by StrawnSC — APPROVED** [review]

**Comment by archidda** [comment]
@jsntcy , @yonzhan , @zhoxing-ms @wangzelin007,
Please help review this. I wanted to ask if we can still check this in for the July milestone.

**Review by jsntcy — APPROVED** [review]

**Comment by jsntcy** [comment]
@archidda, could you please rerun your tests to ensure no regression and add new arguments in tests to ensure new arguments are workable?

**Comment by archidda** [comment]
@jsntcy Can we get this deployed as part of the august milestone, please?

**Comment by archidda** [comment]
> @archidda, could you please rerun your tests to ensure no regression and add new arguments in tests to ensure new arguments are workable?

Ack

**Comment by archidda** [comment]
> @archidda, could you please rerun your tests to ensure no regression and add new arguments in tests to ensure new arguments are workable?

@jsntcy Added the unit test

**Review by wangzelin007 — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py [hunk 1]
```diff
@@ -804,6 +804,10 @@ def load_arguments(self, _):
         c.argument('docker_registry_server_user', options_list=['--docker-registry-server-user', '-d'], help='The container registry server username.')
         c.argument('docker_registry_server_password', options_list=['--docker-registry-server-password', '-w'],
                    help='The container registry server password. Required for private registries.')
+        c.argument('runtime_version', options_list=['--runtime-version'],
+                   help='The runtime version for logic app.', arg_type=get_enum_type(LOGICAPPS_NODE_RUNTIME_VERSIONS))
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/_params.py:808]:
```suggestion
        c.argument('runtime_version', help='The runtime version for logic app.', arg_type=get_enum_type(LOGICAPPS_NODE_RUNTIME_VERSIONS))
```
There is no need to explicitly declare the default option name

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/appservice/_params.py:808]: This is the default mapping rule of CLI core: when the `options_list` is not specified, the option name defaults to the argument name, so you can see that many parameters do not declare `options_list`

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_logicapp_commands.py [hunk 2]
```diff
@@ -107,6 +107,43 @@ def test_logicapp_e2e(self, resource_group):
             JMESPathCheck('length([])', 0)
         ])
 
+
+    @ResourceGroupPreparer(location=DEFAULT_LOCATION)
+    @StorageAccountPreparer()
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_logicapp_commands.py:112]:
```suggestion
    @StorageAccountPreparer(location=DEFAULT_LOCATION)
```

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_logicapp_commands.py [hunk 3]
```diff
@@ -107,6 +107,43 @@ def test_logicapp_e2e(self, resource_group):
...
+    def test_logicapp_e2etest_logicapp_versions_e2e(self, resource_group):
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_logicapp_commands.py:113]:
```suggestion
    def test_logicapp_e2etest_logicapp_versions_e2e(self, resource_group, storage_account):
```

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_logicapp_commands.py [hunk 4]
```diff
@@ -107,6 +107,43 @@ def test_logicapp_e2e(self, resource_group):
...
+        logicapp_name = self.create_random_name(prefix='logic-e2e', length=24)
+        plan = self.create_random_name(prefix='logic-e2e-plan', length=24)
+        storage = self.create_random_name(prefix='logicstorage', length=24)
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_logicapp_commands.py:116]:
```suggestion
```

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_logicapp_commands.py [hunk 5]
```diff
@@ -107,6 +107,43 @@ def test_logicapp_e2e(self, resource_group):
...
+        runtime_version = '~16'
+        functions_version = '4'
+        self.cmd('appservice plan create -g {} -n {} --sku WS1'.format(resource_group, plan)).get_output_in_json()['id']
+        self.cmd('appservice plan list -g {}'.format(resource_group))
+        self.cmd('storage account create --name {} -g {} -l {} --sku Standard_LRS'.format(storage, resource_group, DEFAULT_LOCATION))
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_logicapp_commands.py:121]:
```suggestion
```

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_logicapp_commands.py [hunk 6]
```diff
@@ -107,6 +107,43 @@ def test_logicapp_e2e(self, resource_group):
...
+        self.cmd('logicapp create -g {} -n {} -p {} -s {} --runtime-version {} --functions-version {}'.format(
+            resource_group,
+            logicapp_name,
+            plan,
+            storage,
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_logicapp_commands.py:126]:
```suggestion
            storage_account,
```

**Comment by yanzhudd** [comment]
Hi @archidda 
it seems that the reason of tests failure is not been rerun. May I ask if you have tried to rerun the failed tests in live mode? (add `--live` parameter in `azdev test` command)

**Comment by archidda** [comment]
> > ❌AzureCLI-FullTest
> 
> Hi @archidda it seems that the reason of tests failure is not been rerun. May I ask if you have tried to rerun the failed tests in live mode? (add `--live` parameter in `azdev test` command)

Thank you, I have updated the recordings and now all test runs without error.

**Review by jsntcy — APPROVED** [review]

**Comment by archidda** [comment]
[heart]         Arjun Chiddarwar reacted to your message:
________________________________

**Pull request merged**